src/main/java/uk/gov/hmcts/reform/sendletter/services/ftp/FtpClient.java

```diff
@@ -152,15 +152,12 @@ public void deleteReport(String reportPath) {
         });
     }
 
-    public void deleteFile(String filePath) {
-        runWith(sftp -> {
-            try {
-                sftp.rm(filePath);
-                return null;
-            } catch (Exception exc) {
-                throw new FtpException("Error while deleting file: " + filePath, exc);
-            }
-        });
+    public void deleteFile(String filePath, SFTPClient sftpClient) {
+        try {
+            sftpClient.rm(filePath);
+        } catch (IOException exc) {
```

Comment:
We used to catch all exceptions and now it's `IOException` only. Will we handle runtime exceptions appropriately?

Reply:
Fair doubt right here. So I have scavenged down the line what happens during method call and it is mainly `IOException` but there is a `Promise<T>` thing somewhere there which can be any other exception but is not checked due to type erasure. Will roll back to `Exception` here ;)